fog/view/server/src/fog_view_router_server.rs

```diff
@@ -74,16 +89,48 @@ where
             mc_util_grpc::HealthService::new(Some(readiness_indicator.into()), logger.clone())
                 .into_service();
 
-        // Package service into grpc server
-        log::info!(
-            logger,
-            "Starting Fog View Router server on {}",
-            config.client_listen_uri.addr(),
-        );
-        let router_server_builder = grpcio::ServerBuilder::new(env.clone())
-            .register_service(fog_view_router_service)
-            .register_service(health_service)
-            .bind_using_uri(&config.client_listen_uri, logger.clone());
+        let router_server_builder = match config.client_listen_uri {
+            RouterClientListenUri::Streaming(ref streaming_uri) => {
+                let fog_view_router_service =
+                    view_grpc::create_fog_view_router_api(FogViewRouterService::new(
+                        enclave.clone(),
+                        shards,
+                        config.chain_id.clone(),
+                        client_authenticator,
+                        logger.clone(),
+                    ));
+                log::debug!(logger, "Constructed Fog View Router streaming GRPC Service");
+                log::info!(
+                    logger,
+                    "Starting Fog View Router unary server on {}",
+                    streaming_uri.addr(),
+                );
+                grpcio::ServerBuilder::new(env.clone())
+                    .register_service(fog_view_router_service)
+                    .register_service(health_service)
+                    .bind_using_uri(streaming_uri, logger.clone())
+            }
+            RouterClientListenUri::Unary(ref unary_uri) => {
+                let fog_view_router_service =
+                    view_grpc::create_fog_view_api(FogViewRouterService::new(
+                        enclave.clone(),
+                        shards,
+                        config.chain_id.clone(),
+                        client_authenticator,
+                        logger.clone(),
+                    ));
+                log::debug!(logger, "Constructed Fog View Router unary GRPC Service");
+                log::info!(
+                    logger,
+                    "Starting Fog View Router unary server on {}",
+                    unary_uri.addr(),
+                );
+                grpcio::ServerBuilder::new(env.clone())
+                    .register_service(fog_view_router_service)
+                    .register_service(health_service)
+                    .bind_using_uri(unary_uri, logger.clone())
+            }
+        };
```

Comment:
:wrench: This might be a bit of an extreme refactoring, but maybe some of it is useful.
As currently written it's difficult to see what is intentionally different between the match statements.
At the same time I'm not fond of how the refactoring seems to obfuscate
```suggestion
        let (uri, type_string, function) = match config.client_listen_uri {
            RouterClientListenUri::Streaming(ref streaming_uri) => (streaming_uri, "streaming", view_grpc::create_fog_view_router_api),
            RouterClientListenUri::Unary(ref unary_uri) => (unary_uri, "unary", view_grpc::create_fog_view_api),
        };
        let fog_view_router_service =
            function(FogViewRouterService::new(
                enclave.clone(),
                shards,
                config.chain_id.clone(),
                client_authenticator,
                logger.clone(),
            ));
        log::debug!(logger, "Constructed Fog View Router {} GRPC Service", type_string);
        log::info!(
            logger,
            "Starting Fog View Router {} server on {}", type_string
            uri.addr(),
        );
        let router_server_builder = grpcio::ServerBuilder::new(env.clone())
            .register_service(fog_view_router_service)
            .register_service(health_service)
            .bind_using_uri(uri, logger.clone());
```

Reply:
This won't build because `uri` has different types given the different match arms (one is `&mc_util_uri::Uri<mc_fog_uri::FogViewRouterScheme>` and the other is `&mc_util_uri::Uri<mc_fog_uri::FogViewScheme>`, hence I don't think you can do this (and was why i originally did it this way).